### DISCUSSION TITLE
Unifi install fails with "deb-systemd-invoke" error

### DISCUSSION THREAD

**Discussion by triggad23** [tteck/Proxmox#2797] [Help Needed]
Hi,

I'm trying to install unifi with 

_bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/unifi.sh)"_

I tried many times with different settings, updated the proxmox host to 8.1.10, but the error keeps the same:

_Could not execute systemctl:  at /usr/bin/deb-systemd-invoke line 145._

After this the skript states success, but the controller is stucked at

_UniFi Network Application is starting up...
Please wait a moment_

Here are the settings I chose:

> Using Advanced Settings
> Using Distribution: debian
> Using debian Version: 12
> Using Container Type: 1
> Using Root Password: password1234
> Container ID: 101
> Using Hostname: unifi
> Using Disk Size: 8
> Allocated Cores: 2
> Allocated RAM: 2048
> Using Bridge: vmbr0
> Using IP Address: dhcp
> Using Gateway IP Address: Default
> Using APT-Cacher IP Address: Default
> Disable IPv6: no
> Using Interface MTU Size: Default
> Using DNS Search Domain: Host
> Using DNS Server IP Address: Host
> Using Vlan: Default
> Enable Root SSH Access: no
> Enable Verbose Mode: yes
> Creating a Unifi LXC using the above advanced settings

Greets
Daniel

**Comment by tteck** [comment]
🏁 The script runs without any problems from this end.

```
   __  __      _ _____ 
  / / / /__   (_) __(_)
 / / / / __ \/ / /_/ / 
/ /_/ / / / / / __/ /  
\____/_/ /_/_/_/ /_/   
 
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 110
Using Hostname: unifi
Using Disk Size: 8
Allocated Cores: 2
Allocated RAM: 2048
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Using APT-Cacher IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Unifi LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using thin-pool for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 110 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.250 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.114.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed Eclipse Temurin JRE
 ✓ Installed MongoDB
 ✓ Installed UniFi Network Server
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!

Unifi should be reachable by going to the following URL.
         https://192.168.86.250:8443 
```

![Screenshot 2024-04-16 6 19 44 AM](https://github.com/tteck/Proxmox/assets/21342670/37e6f6c6-550f-4d23-85b4-5d7eb711fede)

**Comment by triggad23** [comment]
hmm, strange... Thanks for your answer!

I tried again with default settings and still the same...

What else can this be? could there be a problem with my proxmox server? I installed this host a few weeks ago from scratch.

This is line 145 from the deb-systemd-invoke file:

           _system('systemctl', '--quiet', @instance_args, $action, @start_units) == 0 or die("Could not execute systemctl: $!");_

it seems he tries to write an error message here...

Perhaps I should try to reinstall the proxmox... 
Is there a way without loosing all VMs? I have no Backup target in this network at the moment...

Greets
Daniel

**Reply by tteck** [reply]
Are executing the script as user `root`?

**Reply by triggad23** [reply]
I'm just using the shell window in the proxmox Web UI and I'm logged in as root.

**Comment by triggad23** [comment]
Hi,
I made a fresh proxmox install and tried the unifi script as second thing (first was setting up a firewall to gain internet access).

But the error still keeps the same...

Any other ideas?

**Reply by tteck** [reply]
Still runs ok here

```
   __  __      _ _____ 
  / / / /__   (_) __(_)
 / / / / __ \/ / /_/ / 
/ /_/ / / / / / __/ /  
\____/_/ /_/_/_/ /_/   
 
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: tommy
Container ID: 110
Using Hostname: unifi
Using Disk Size: 8
Allocated Cores: 2
Allocated RAM: 2048
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Using APT-Cacher IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Unifi LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using thin-pool for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 110 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.250 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.114.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed Eclipse Temurin JRE
 ✓ Installed MongoDB
 ✓ Installed UniFi Network Server
 ✓ Cleaned
 ✓ Completed Successfully!

Unifi should be reachable by going to the following URL.
         https://192.168.86.250:8443 
```

**Comment by reaper1412** [comment]
I experience the same thing.
Would be great if anyone have a solution.

Using Debian 12 as the LXC Container.
Running Proxmox 8.2.2

**Reply by tteck** [reply]
Yet another check.
🏁 The script runs without any problems from this end.

```
   __  __      _ _____ 
  / / / /__   (_) __(_)
 / / / / __ \/ / /_/ / 
/ /_/ / / / / / __/ /  
\____/_/ /_/_/_/ /_/   
 
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 112
Using Hostname: unifi
Using Disk Size: 8
Allocated Cores: 2
Allocated RAM: 2048
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Using APT-Cacher IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Unifi LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using thin-pool for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 112 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.250 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.112.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed Eclipse Temurin JRE
 ✓ Installed MongoDB
 ✓ Installed UniFi Network Server
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!

Unifi should be reachable by going to the following URL.
         https://192.168.86.250:8443 
```

**Reply by reaper1412** [reply]
Hey thank you for the quick reply!
I manage to install the Unifi controller by using your Debian 12 LXC script to create an empty Debian LXC Container,
then after it is running, I run the Unifi easy install script from Glenn R.
Now my unify controller is running.
I don't know what went wrong there, but that's how I solved it.

**Reply by tteck** [reply]
Remember, the update instructions listed in the repository will _probably_ not work for you.

**Reply by reaper1412** [reply]
Thanks for the reminder.
I just want to get it up and running quickly now.
I'll try again your script another time.

**Comment by MarvinMynx** [comment]
I experience the same thing.
Would be great if anyone have a solution.

Using Debian 12 as the LXC Container.
Running Proxmox Virtual Environment 8.2.4

During Installation it seems to be stuck at a specific point for minutes, until it just continues. see Screenshots for details. 
First here are the used Settings:
![Screenshot 2024-07-16 124057](https://github.com/user-attachments/assets/8cec2dbc-520c-471d-97bd-1fb5f77dccd9)

The Timestamps relatively accurately describes the time between the "stuck" state and when it continues. 
This can also be seen in the memory and CPU usage graphs of Proxmox (or basically in any graph, but in memory graph it is seen best i think)
(I'm not a Developer so sorry for my bad terminology and way of debugging and showing logs and informations, just trying to help) 
![Screenshot 2024-07-16 122714](https://github.com/user-attachments/assets/c511596d-7b49-4b87-8271-99301ade7061)
![Screenshot 2024-07-16 123901](https://github.com/user-attachments/assets/fa532fca-27b8-410f-bbfe-7a7ae590013d)

If i search the Log in the Screenshots i see this Line exactly after the "stuck" state. 
`\Could not execute systemctl:  at /usr/bin/deb-systemd-invoke line 145.`
Everything after this happens in just a under a second for me.

The Memory Usage actually increases during the "Stuck" state but just minimal, while every other usage/load on the system stays "calm"
![Screenshot 2024-07-16 123916](https://github.com/user-attachments/assets/8e0dee22-ebae-40ae-947a-aee2dc5f1cae)
![Screenshot 2024-07-16 123923](https://github.com/user-attachments/assets/112a62ab-fdfe-40c8-a3e2-b6dbcc6fe445)

I opened a shell of the created LXC after creation and just logged in to show open ports and i don't see any UniFi stuff or anything on port 8443. 
![Screenshot 2024-07-16 123952](https://github.com/user-attachments/assets/6b77b011-7702-4ccc-8312-0718ac6da7f9)

So there is definitely something going wrong here as it even do not work with default settings (this can actually be seen in all of the screenshots above too, because this was my first try with the same Containter ID so the Graph continue to show here), it fails exactly the same way while it claims to be successful.

**Reply by tteck** [reply]
🔁 If I cannot reproduce the problem on my side, I cannot address an issue that appears to be functioning correctly here.

```
   __  __      _ _____ 
  / / / /__   (_) __(_)
 / / / / __ \/ / /_/ / 
/ /_/ / / / / / __/ /  
\____/_/ /_/_/_/ /_/   
 
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 111
Using Hostname: unifi-july-17-2024
Using Disk Size: 8
Allocated Cores: 2
Allocated RAM: 2048
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Using APT-Cacher IP Address: Default
Disable IPv6: no
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Unifi LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using thin-pool for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 111 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.250 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.114.3
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed Eclipse Temurin JRE
 ✓ Installed MongoDB
 ✓ Installed UniFi Network Server
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!

Unifi should be reachable by going to the following URL.
         https://192.168.86.250:8443 
```

**Reply by reaper1412** [reply]
> I experience the same thing. Would be great if anyone have a solution.

@MarvinMynx are you upgrading your Proxmox Instalation from an older Proxmox version like 7.x?
For me it is like this and I suspect the problem comes from there. Something wrong from the update process.
I want to do a clean Proxmox install and test again the script but still does not have the time for that.

**Reply by MarvinMynx** [reply]
> > I experience the same thing. Would be great if anyone have a solution.
> 
> @MarvinMynx are you upgrading your Proxmox Instalation from an older Proxmox version like 7.x? For me it is like this and I suspect the problem comes from there. Something wrong from the update process. I want to do a clean Proxmox install and test again the script but still does not have the time for that.

@reaper1412  I think i needed to clean install on the Release of Proxmox 8.x for other reasons anyway.
So i don't think a clean install of Proxmox will Resolve the issue.

**Reply by reaper1412** [reply]
> @reaper1412 I think i needed to clean install on the Release of Proxmox 8.x for other reasons anyway. So i don't think a clean install of Proxmox will Resolve the issue.

Keep us posted when you did your clean install.

**Reply by MarvinMynx** [reply]
> > @reaper1412 I think i needed to clean install on the Release of Proxmox 8.x for other reasons anyway. So i don't think a clean install of Proxmox will Resolve the issue.
> 
> Keep us posted when you did your clean install.

@reaper1412 Well I'm not going to clean install *again*? The sentence you quoted is past tense. That's why i don't think a clean install of Proxmox will Resolve the issue. And frankly i have neither the time nor the money (Hardware/Environment) to test stuff like that.

**Reply by MarvinMynx** [reply]
> If I cannot reproduce the problem on my side, I cannot address an issue that appears to be functioning correctly here.

@tteck you could try to work with us, to fix this? Because not everyone is running your specific Hardware.

**Reply by tteck** [reply]
What would you like me to do?

**Reply by MarvinMynx** [reply]
> What would you like me to do?

@tteck well: what would you do if you got the Error Messages we described here in your tests? (yes this is gonna take some time, but I'm up for it to check for new comments multiple times a day (don't know your time zone but seems pretty similar) and give answers of my system config etc.)

Maybe we could even get in contact, for easier debugging? Like Discord, Signal or what ever?

**Reply by tteck** [reply]
> what would you do if you got the Error Messages we described here in your tests?

I would research why my system errors when trying to use `systemctl` through the `deb-systemd-invoke` script during a package installation or upgrade process.

**Reply by MarvinMynx** [reply]
> > what would you do if you got the Error Messages we described here in your tests?
> 
> I would research why my system errors when trying to use `systemctl` through the `deb-systemd-invoke` script during a package installation or upgrade process.

@tteck how would you do that? cat / tail / less /var/log/messages? or journalctl? which unit if? Or which Log files or applications would you use to research why my system errors when trying to use `systemctl` through the `deb-systemd-invoke` script during a package installation or upgrade process?

**Reply by MarvinMynx** [reply]
i cloned your Repo and researched your install scripts. In `install\unifi-install.sh` i checked all three gpg and Repo Files. They where Okay. Then i just tried to do an `apt-get Update`. It Bailed out with the two following messages:
```
E: Repository 'https://dl.ui.com/unifi/debian stable InRelease' changed its 'Codename' value from 'unifi-8.2' to 'unifi-8.3'
N: This must be accepted explicitly before updates for this repository can be applied. See apt-secure(8) manpage for details.
```
Then i used `apt-get update --allow-releaseinfo-change` which got me rid of the Second one but still bailed out with the same E: Message.
The First one i got rid of by using `apt-get clean` first, then `apt-get update --allow-releaseinfo-change` and then i installed all the Packages of the Script which where `apt-get install -y curl sudo mc apt-transport-https gnupg temurin-17-jre mongodb-org unifi`.
Just the `unifi` Package where still missing so this command tried to install it which seems to be stuck at this step `Setting up unifi (8.3.32-26237-1) ...` and then it faild with the same Error message i got running your Script which is `Could not execute systemctl:  at /usr/bin/deb-systemd-invoke line 145.` 

On further researching i noticed that the Unifi Service is present and enabled, but i can't start it. Using `systemctl start unifi` "stucks" the System again for a while and then it Errors out with:
```
Job for unifi.service failed because a timeout was exceeded.
See "systemctl status unifi.service" and "journalctl -xeu unifi.service" for details.
```
and `journalctl -xeu unifi.service` gave me: 
```
Jul 19 23:01:14 unifi systemd[1]: unifi.service: start operation timed out. Terminating.
Jul 19 23:01:14 unifi systemd[1]: unifi.service: Failed with result 'timeout'.
-- Subject: Unit failed
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- The unit unifi.service has entered the 'failed' state with result 'timeout'.
Jul 19 23:01:14 unifi systemd[1]: Failed to start unifi.service - unifi.
-- Subject: A start job for unit unifi.service has failed
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- A start job for unit unifi.service has finished with a failure.
-- 
-- The job identifier is 490 and the job result is failed.
Jul 19 23:01:14 unifi systemd[1]: unifi.service: Consumed 9.815s CPU time.
-- Subject: Resources consumed by unit runtime
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- The unit unifi.service completed and consumed the indicated resources.
Jul 19 23:01:14 unifi systemd[1]: unifi.service: Scheduled restart job, restart counter is at 2.
-- Subject: Automatic restarting of a unit has been scheduled
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- Automatic restarting of the unit unifi.service has been scheduled, as the result for
-- the configured Restart= setting for the unit.
Jul 19 23:01:14 unifi systemd[1]: Stopped unifi.service - unifi.
-- Subject: A stop job for unit unifi.service has finished
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- A stop job for unit unifi.service has finished.
-- 
-- The job identifier is 612 and the job result is done.
Jul 19 23:01:14 unifi systemd[1]: unifi.service: Consumed 9.815s CPU time.
-- Subject: Resources consumed by unit runtime
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- The unit unifi.service completed and consumed the indicated resources.
Jul 19 23:01:14 unifi systemd[1]: Starting unifi.service - unifi...
-- Subject: A start job for unit unifi.service has begun execution
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- A start job for unit unifi.service has begun execution.
-- 
-- The job identifier is 612.
Jul 19 23:01:14 unifi unifi[5825]: init complete...
Jul 19 23:01:14 unifi unifi-network-service-helper[5825]: <13>Jul 19 23:01:14 unifi: init complete...
Jul 19 23:01:14 unifi unifi[5829]: Skipping init-uos...
Jul 19 23:01:14 unifi unifi-network-service-helper[5829]: <13>Jul 19 23:01:14 unifi: Skipping init-uos...

```

After researching even further i in unifi server Logs i think i narrowed it down to a Problem with MongoDB at which point I'm stuck and i dont know how to research further.

```
root@unifi:/etc/apt/sources.list.d# cat /usr/lib/unifi/logs/server.log
[...]
[2024-07-19 23:17:55,114] <mongo-db> WARN  mongo  - Stop listening to Mongo logs after process has exited
[2024-07-19 23:17:55,114] <mongo-db> INFO  mongo  - Database process stopped, code=132
[2024-07-19 23:17:58,115] <mongod-tail> WARN  mongo  - Mongo log not found during startup
[2024-07-19 23:17:58,124] <mongo-db> WARN  mongo  - Stop listening to Mongo logs after process has exited
[2024-07-19 23:17:58,124] <mongo-db> INFO  mongo  - Database process stopped, code=132
[2024-07-19 23:18:01,125] <mongod-tail> WARN  mongo  - Mongo log not found during startup
[2024-07-19 23:18:01,134] <mongo-db> WARN  mongo  - Stop listening to Mongo logs after process has exited
[2024-07-19 23:18:01,134] <mongo-db> INFO  mongo  - Database process stopped, code=132
[2024-07-19 23:18:04,135] <mongod-tail> WARN  mongo  - Mongo log not found during startup
[2024-07-19 23:18:04,143] <mongo-db> WARN  mongo  - Stop listening to Mongo logs after process has exited
[2024-07-19 23:18:04,144] <mongo-db> INFO  mongo  - Database process stopped, code=132
[2024-07-19 23:18:07,144] <mongod-tail> WARN  mongo  - Mongo log not found during startup
[2024-07-19 23:18:07,153] <mongo-db> WARN  mongo  - Stop listening to Mongo logs after process has exited
[2024-07-19 23:18:07,153] <mongo-db> INFO  mongo  - Database process stopped, code=132
[2024-07-19 23:18:10,153] <mongod-tail> WARN  mongo  - Mongo log not found during startup
[2024-07-19 23:18:10,162] <mongo-db> WARN  mongo  - Stop listening to Mongo logs after process has exited
[2024-07-19 23:18:10,162] <mongo-db> INFO  mongo  - Database process stopped, code=132
[2024-07-19 23:18:13,163] <mongod-tail> WARN  mongo  - Mongo log not found during startup
[2024-07-19 23:18:13,172] <mongo-db> WARN  mongo  - Stop listening to Mongo logs after process has exited
[2024-07-19 23:18:13,172] <mongo-db> INFO  mongo  - Database process stopped, code=132
[2024-07-19 23:18:16,173] <mongod-tail> WARN  mongo  - Mongo log not found during startup
[2024-07-19 23:18:16,187] <mongo-db> WARN  mongo  - Stop listening to Mongo logs after process has exited
[2024-07-19 23:18:16,187] <mongo-db> INFO  mongo  - Database process stopped, code=132
[2024-07-19 23:18:19,187] <mongod-tail> WARN  mongo  - Mongo log not found during startup
[2024-07-19 23:18:19,196] <mongo-db> WARN  mongo  - Stop listening to Mongo logs after process has exited
[2024-07-19 23:18:19,197] <mongo-db> INFO  mongo  - Database process stopped, code=132
root@unifi:/etc/apt/sources.list.d# mongo -port 27017 --nodb
-bash: mongo: command not found
root@unifi:/etc/apt/sources.list.d# mongo
mongod        mongoexport   mongoimport   mongos        mongostat     
mongodump     mongofiles    mongorestore  mongosh       mongotop      
root@unifi:/etc/apt/sources.list.d# mongod -port 27017 --nodb
Illegal instruction
root@unifi:/etc/apt/sources.list.d# mongos -port 27017 --nodb
Illegal instruction
root@unifi:/etc/apt/sources.list.d# mongosh -port 27017 --nodb
Current Mongosh Log ID: 669ad8418a621040a4482f8a
Using Mongosh:          2.2.12

For mongosh info see: https://docs.mongodb.com/mongodb-shell/


To help improve our products, anonymous usage data is collected and sent to MongoDB periodically (https://www.mongodb.com/legal/privacy-policy).
You can opt-out by running the disableTelemetry() command.

> use ace
MongoshInvalidInputError: [SHAPI-10004] No connected database
> show dbs
MongoshInvalidInputError: [SHAPI-10004] No connected database
> exit
root@unifi:/etc/apt/sources.list.d# ss -tulpn
Netid   State    Recv-Q   Send-Q     Local Address:Port       Peer Address:Port   Process                                                  
udp     UNCONN   0        0                0.0.0.0:68              0.0.0.0:*       users:(("dhclient",pid=90,fd=7))                        
tcp     LISTEN   0        100            127.0.0.1:25              0.0.0.0:*       users:(("master",pid=349,fd=13))                        
tcp     LISTEN   0        4096                   *:22                    *:*       users:(("sshd",pid=178,fd=3),("systemd",pid=1,fd=56))   
tcp     LISTEN   0        100                [::1]:25                 [::]:*       users:(("master",pid=349,fd=14))                        
root@unifi:/etc/apt/sources.list.d# systemctl status mongod.service 
* mongod.service - MongoDB Database Server
     Loaded: loaded (/lib/systemd/system/mongod.service; disabled; preset: enabled)
     Active: inactive (dead)
       Docs: https://docs.mongodb.org/manual
root@unifi:/etc/apt/sources.list.d# systemctl start mongod.service 
root@unifi:/etc/apt/sources.list.d# systemctl status mongod.service 
x mongod.service - MongoDB Database Server
     Loaded: loaded (/lib/systemd/system/mongod.service; disabled; preset: enabled)
     Active: failed (Result: signal) since Fri 2024-07-19 23:19:33 CEST; 1s ago
   Duration: 10ms
       Docs: https://docs.mongodb.org/manual
    Process: 7440 ExecStart=/usr/bin/mongod --config /etc/mongod.conf (code=killed, signal=ILL)
   Main PID: 7440 (code=killed, signal=ILL)
        CPU: 9ms

Jul 19 23:19:33 unifi systemd[1]: Started mongod.service - MongoDB Database Server.
Jul 19 23:19:33 unifi systemd[1]: mongod.service: Main process exited, code=killed, status=4/ILL
Jul 19 23:19:33 unifi systemd[1]: mongod.service: Failed with result 'signal'.
root@unifi:/etc/apt/sources.list.d# journalctl -u mongod.service
Jul 19 23:19:33 unifi systemd[1]: Started mongod.service - MongoDB Database Server.
Jul 19 23:19:33 unifi systemd[1]: mongod.service: Main process exited, code=killed, status=4/ILL
Jul 19 23:19:33 unifi systemd[1]: mongod.service: Failed with result 'signal'.
root@unifi:/etc/apt/sources.list.d# ls -alFh /usr/lib/unifi/data/db/
total 13K
drwxr-xr-x 2 unifi unifi 2 Jul 16 12:23 ./
drwxr-x--- 4 unifi unifi 7 Jul 19 23:16 ../
root@unifi:/etc/apt/sources.list.d# cd  /var/logs/unif  
-bash: cd: /var/logs/unif: No such file or directory
root@unifi:/etc/apt/sources.list.d# cd /var/lo       
local/ lock/  log/   
root@unifi:/etc/apt/sources.list.d# cd /var/lo
local/ lock/  log/   
root@unifi:/etc/apt/sources.list.d# cd /var/log/
apt/     journal/ mongodb/ private/ runit/   unifi/   
root@unifi:/etc/apt/sources.list.d# cd /var/log/
apt/     journal/ mongodb/ private/ runit/   unifi/   
root@unifi:/etc/apt/sources.list.d# cd /var/log/unifi/
root@unifi:/var/log/unifi# ls
gc.log  gc.log.0  gc.log.1  hotspot.log  inform_request.log  migration.log  server.log  startup.log  state.log  tasks.log
root@unifi:/var/log/unifi# ls -alFht
total 1.9M
-rw-r--r-- 1 unifi unifi 1.2M Jul 19 23:21 server.log
-rw-r--r-- 1 unifi unifi  334 Jul 19 23:19 gc.log
-rw-r--r-- 1 unifi unifi   72 Jul 19 23:16 startup.log
drwxr-x--- 2 unifi unifi   12 Jul 19 23:16 ./
-rw-r--r-- 1 unifi unifi 504K Jul 19 23:16 gc.log.0
-rw-r--r-- 1 unifi unifi  522 Jul 19 23:00 gc.log.1
drwxr-xr-x 8 root  root    18 Jul 19 22:30 ../
-rw-r--r-- 1 unifi unifi    0 Jul 16 12:23 inform_request.log
-rw-r--r-- 1 unifi unifi    0 Jul 16 12:23 hotspot.log
-rw-r--r-- 1 unifi unifi    0 Jul 16 12:23 tasks.log
-rw-r--r-- 1 unifi unifi    0 Jul 16 12:23 migration.log
-rw-r--r-- 1 unifi unifi    0 Jul 16 12:23 state.log
root@unifi:/var/log/unifi# 
```

Is anything of this useful or could you assist with further researching? 

(Finished Editing this Comment at Jul 19, 2024, @11:30 PM GMT+2)

**Reply by tteck** [reply]
What is proxmox.fritz.box?

**Reply by MarvinMynx** [reply]
> What is proxmox.fritz.box?

@tteck well that's my Proxmox Host/Hypervisor. Equivalent to what you described as "your end" and "this end". 
But from where did you get this? I Never mentioned this Local Domain nor this FQDN here.
EDIT: Oh its in the First Screenshot's Firefox Address bar never mind.

**Reply by MarvinMynx** [reply]
@tteck you mentioned that AVX Instruction set is needed. I just checked that MongoDB need that specific instruction set so it seems. How would you check if your Proxmox Host has this Feature? `cat /proc/cpuinfo | grep -i "avx"` `grep -o 'avx[^ ]*' /proc/cpuinfo` If so than my Host/Hypervisor CPU doesn't support it or it's disable in the BIOS/UEFI if this is Possible?
@triggad23 and @reaper1412  what's with your Host/Hypervisor CPU's does they Support AVX Instruction Sets?

@tteck Maybe you could include a check in the script to Error Out with a Specific Warning if AVX Instruction set can't be found? Just Error Out with the Message "🛈 CPU with AVX Instruction Set required" if `grep -o 'avx[^ ]*' /proc/cpuinfo ` would count 0 Lines, or something like that. i don't know. But would increase user experience.

This is my CPU, i don't know how to check for AVX. It seems like Intel won't show Instructionsets Of this CPU at all.
Intel® Pentium® Gold Prozessor G5420 
https://ark.intel.com/content/www/us/en/ark/products/135456/intel-pentium-gold-g5420-processor-4m-cache-3-80-ghz.html

**Reply by reaper1412** [reply]
> @triggad23 and @reaper1412 what's with your Host/Hypervisor CPU's does they Support AVX Instruction Sets?

My host use an Intel Celeron N5105 Processor. No AVX here.
For me the script does not work. But when I install manually, it works.

> @tteck you mentioned that AVX Instruction set is needed.

Maybe thats why the script does not work on my part.

**Reply by MarvinMynx** [reply]
> Maybe thats why the script does not work on my part.

That's definitely it. I just found out that i can't enable it, my CPU seems to not support it at all.
![Unbenannt](https://github.com/user-attachments/assets/b2943a99-1e71-440a-8bc7-3244b3c1e548)

Well i would suggest it's the same with @reaper1412 which makes us 3 "visible" individuals who where running into this "trap" even so it says so on the Website in Red Letters.
@tteck i would suggest a "🛈 CPU with AVX Instruction Set required" Error Message on Script start if no AVX Instruction Set is found to avoid unnecessary  waiting and deploying and debugging time, tickets and unhappy user.

@tteck Or Edit your Script in a way that makes AVX Instruction Sets unnecessary in a way like the `Unifi easy install script` from `Glenn R.` seems to be doing.

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/commit/2febf2c39bbdca74507a864b632908f4c14a8265

**Reply by tteck** [reply]
The Glenn R script is probably using an old version of MongoDB that is beyond the End of Life Date.
https://www.mongodb.com/legal/support-policy/lifecycles

**Reply by MarvinMynx** [reply]
> The Glenn R script is probably using an old version of MongoDB that is beyond the End of Life Date. https://www.mongodb.com/legal/support-policy/lifecycles

ah bummer :/

**Reply by MarvinMynx** [reply]
> [2febf2c](https://github.com/tteck/Proxmox/commit/2febf2c39bbdca74507a864b632908f4c14a8265)

it Works:

![Screenshot 2024-07-20 123811](https://github.com/user-attachments/assets/38f40153-43d6-4c63-a1bc-d8e76f369c6e)

**Reply by MarvinMynx** [reply]
> > The Glenn R script is probably using an old version of MongoDB that is beyond the End of Life Date. https://www.mongodb.com/legal/support-policy/lifecycles
> 
> ah bummer :/

i think i just confirmed it, using his Changelog. 
I Marked the Line with > chars at the beginning and < chars at the end of the Line.
He is Using MongoDB Version 4.x which is out of it's lifecycle since round about 2 Years.

```
 # UniFi Network Application Installation Script changelog

 # Author   | Glenn Rietveld
 # Email    | glennrietveld8@hotmail.nl
 # Website  | https://GlennR.nl

####################
#                                    #
#         Changelog           #
#                                    #
####################

 |  UniFi Network Application Installation Script  |  Version 7.7.1
 
 - Fixed an issue where the incorrect MongoDB shell is used for older MongoDB versions if mongosh is installed.
 - Decreases minimum space check for /var/log to 25MB.
 
 ####

 |  UniFi Network Application Installation Script  |  Version 7.7.0
 
 - Added sha256sum check when downloading the UniFi Network Application.
 - Improved setting MongoDB commands.
 - Improved support files.
 
 ####

 |  UniFi Network Application Installation Script  |  Version 7.6.9
 
 - Added support for detecting UniFi-Native setups and redirect them to the Easy Update Script.
 - Fixed inability to install Glenn R. Compiled MongoDB dependencies setups without the default repositories.
 
 ####

 |  UniFi Network Application Installation Script  |  Version 7.6.8
 
 - Added secondary method to get CPU model.
 - Prevent setting incorrect max supported MongoDB version if CPU model is empty.
 
 ####

 |  UniFi Network Application Installation Script  |  Version 7.6.7
 
 - Added support for the HTTP protocol for the Glenn R. Compiled MongoDB.
 - Improved logging for downloading repository keys.
 - Improved support files.
 - Improved setting script directory on UniFi OS Consoles.
 - Fixed an issue where the script would fail if apt-listbugs is installed.
 - Fixed an issue where grep errors could be shown.
 
 ####

 |  UniFi Network Application Installation Script  |  Version 7.6.6
 
 - Allow users to still choose to upload support files during aborts despite originally choosing no earlier.
 - Improved the script for UniFi OS Consoles.
 - Fixed an issue where the MongoDB unsupported version change process didn't always properly re-install UniFi.
 - Fixed an issue with removing MongoDB packages during the MongoDB unsupported version change process.
 
 ####

 |  UniFi Network Application Installation Script  |  Version 7.6.5
 
 - Added UniFi Network Native Application uninstallation process if users attempted to install it on the incorrect devices.
 - Improved English language detection.
 - Improved adding DEB822 repositories.
>>>>>>>>>>>>>> - Improved adding MongoDB 4.0 repositories.  <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
>>>>>>>>>>>>>>  - Improved arm64 setup recovery if users installed unsupported MongoDB 7.0.  <<<<<<<<<<<
 - Improved the MongoDB unsupported version change process.
 - Fixed an issue where errors could be printed when setting language to English.
 - Fixed incorrect colors for Failed/Success messages.
 - Fixed an issue where the script could install incorrect MongoDB versions while performing an installation recovery.
```

EDIT: 22.07.2024 (DD.MM.YYYY): Well to be fair, he is using 4.4 which is EOL since February 2024.
Anyone who still want toi use this, just install an Debian LXC (bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/debian.sh)") with  2GB RAM - 8GB Storage - 2vCPU and then login to that LXC and run this Command: curl -sO https://get.glennr.nl/unifi/install/install_latest/unifi-latest.sh && bash unifi-latest.sh
that should be it.
